### PR TITLE
Support flushing of pending requests and verifying none are outstanding

### PR DESCRIPTION
**Pull request by nathanboktae**
Builds on #21, and brings this library to near parity with `$httpBackend`.

### CODE REVIEW

File: lib/pending.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    if (pendingRequests[i].request === request) {
+      clearTimeout(pendingRequests[i].timeoutId);
+      pendingRequests.splice(i, 1);
+      return;
+    }
+  }
+};
+
+pending.clear = function () {
+  pendingRequests.length = 0;
```
Comment (callmehiphop) [inline lib/pending.js]: Should be calling `clearTimeout` on any of the remaining pending requests here as well?

File: lib/pending.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+};
+
+/**
+ * Return any outstanding pending requests
+ * @return {array[mock]}
+ */
+pending.outstanding = function() {
+  var requests = [];
+  _.each(pendingRequests, function(pending) {
```
Comment (callmehiphop) [inline lib/pending.js]: We could slim this one down if we used `_.map()` instead

File: lib/pending.js (hunk 1 above)
Comment (nathanboktae) [inline lib/pending.js]: Yes, good catch.

**Comment by callmehiphop** [comment]
Can you rebase this please?

**Comment by nathanboktae** [comment]
> Can you rebase this please?

Yup done.

File: index.js [hunk 3]
```diff
@@ -51,18 +52,37 @@ _.each(['GET', 'POST', 'PUT', 'DELETE', 'PATCH'], function(method) {
...
   }
 };
 
+/**
+ * Assert that there are no stubbed requests that are not resolved yet.
+ */
+backend.verifyNoOutstandingRequest = function() {
+  var inFlight = pending.outstanding();
+  if (inFlight.length) {
+    throw new Error('Expected no outstanding requests, but there were ' + inFlight.length + '\n' + inFlight[0].toString());
```
Comment (callmehiphop) [inline index.js]: Could we format this to be less than 80 characters per line?

Comment (nathanboktae) [inline index.js]: how's this?

```javascript
    throw new Error('Expected no outstanding expectations, but there were ' +
      expected.length + '\n' + expected[0].toString());
```

Comment (callmehiphop) [inline index.js]: perfect!

**Comment by nathanboktae** [comment]
@callmehiphop I rebased this. Do you need anything else? I still have another PR where I discovered passthrough requests don't work at all :unamused: I'll put that one together after all the others get accepted.

**Comment by callmehiphop** [comment]
This one looks good thanks!

**Pull request merged**